src/app/components/core/configuration-manager/list-type-configuration/list-type-configuration.component.ts

```diff
@@ -10,25 +11,35 @@ import { CustomValidator } from '../../../../directives/custom-validator';
 })
 export class ListTypeConfigurationComponent implements OnInit {
   @Input() configuration;
-  @Output() formState = new EventEmitter<boolean>();
-  form: FormGroup;
+  // @Output() formState = new EventEmitter<boolean>();
+  // form: FormGroup;
   listItemsForm: FormGroup;
+  @Input() group: string = '';
+  @Input() from = '';
+  @Output() changedConfig = new EventEmitter<any>();
+  @Output() formStatusEvent = new EventEmitter<any>();
+  initialProperties = [];
 
   constructor(
     public cdRef: ChangeDetectorRef,
-    private rootFormGroup: FormGroupDirective,
+    // private rootFormGroup: FormGroupDirective,
     private fb: FormBuilder) {
     this.listItemsForm = this.fb.group({
       listItems: this.fb.array([])
     })
   }
 
   ngOnInit() {
-    this.form = this.rootFormGroup.control;
+    // this.form = this.rootFormGroup.control;
     let values = this.configuration?.value ? this.configuration.value : this.configuration.default;
     values = JSON.parse(values) as [];
     values.forEach(element => {
-      this.initListItem(element);
+      if(this.configuration.items == 'object') {
+        this.initObjectListItem(element);
+      }
```

Comment:
this check can be moved into the this.initListItem(element) method. By doing that the same check in the  addListItem() can be removed.